bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,172 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class with utility functions to support Wake On Lan (WOL)
+ *
+ * @author Arjan Mels - Initial contribution
+ * @author Sebastian Prehn - Modification to getMACAddress
+ *
+ */
+@NonNullByDefault
+public class WakeOnLanUtility {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WakeOnLanUtility.class);
+    private static final Pattern MAC_REGEX = Pattern.compile("(([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2})");
+
+    private static String COMMAND;
+    static {
+        String os = System.getProperty("os.name").toLowerCase();
+        LOGGER.debug("os: {}", os);
+        if ((os.indexOf("win") >= 0)) {
+            COMMAND = "arp -a %s";
+        } else if ((os.indexOf("mac") >= 0)) {
+            COMMAND = "arp %s";
+        } else { // linux, docker
+            COMMAND = "arping -r -c 1 -C 1 %s";
+        }
+    }
+
+    /**
+     * Get MAC address for host
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static @Nullable String getMACAddress(String hostName) {
+        String cmd = String.format(COMMAND, hostName);
+        LOGGER.debug("cmd: {}", cmd);
+
+        try {
+            Process proc = Runtime.getRuntime().exec(cmd);
+            int returnCode = proc.waitFor();
+            String s;
+            StringBuilder builder = new StringBuilder();
+            BufferedReader input = new BufferedReader(
+                    new InputStreamReader(returnCode == 0 ? proc.getInputStream() : proc.getErrorStream()));
+            while ((s = input.readLine()) != null) {
+                builder.append(s);
+            }
+
+            if (returnCode != 0) {
+                LOGGER.debug("{} failed with error: {}", cmd, builder.toString());
+            } else {
+                Matcher matcher = MAC_REGEX.matcher(builder.toString());
+                String macAddress = null;
+
+                while (matcher.find()) {
+                    String group = matcher.group();
+
+                    if (group.length() == 17) {
+                        macAddress = group;
+                    }
```

Comment:
Shouldn't you break at this point? Or do you always want the last mac address?

Reply:
you are right. can break at this point.